meshtastic/interfaces/ble/lifecycle_disconnect_runtime.py

```
@@ -0,0 +1,490 @@
+            if previous_client is not None
+            else None
+        )
+        if not address_for_registry or address_for_registry == "unknown":
+            address_for_registry = address if address != "unknown" else iface.address
+        addr_disconnect_key = _addr_key(address_for_registry)
+        return (
+            iface._sorted_address_keys(addr_disconnect_key, alias_key),
+            should_schedule_reconnect,
+        )
+
+    def _resolve_disconnect_target(
```

Ruff (0.15.6)

[warning] Too many return statements (7 > 6)

(PLR0911)